### ISSUE TITLE
[Bug] The behavior of starting Dragonfly project is different with the document

### ISSUE THREAD

**Issue by fengzixu** [dragonflyoss/Dragonfly#921]
### Ⅰ. Issue Description

The actual download image behavior is inconsistent with the document description

### Ⅱ. Describe what happened

I try to use Dragonfly according to document [QuickStart](https://github.com/dragonflyoss/Dragonfly/blob/master/docs/quick_start/README.md )

When I performed the Step 5, I didn't see any output. But I found the file(/root/.small-dragonfly/logs/dfclient.log) appears in my host machine, not within the container of dfclient.

```
root@ubuntu-s-1vcpu-1gb-sgp1-01:~/.small-dragonfly# ll
total 24
drwxr-xr-x 6 root root 4096 Sep 17 14:17 ./
drwx------ 6 root root 4096 Sep 17 14:17 ../
drwxr-xr-x 2 root root 4096 Sep 17 14:17 data/
drwxr-xr-x 3 root root 4096 Sep 17 14:16 dfdaemon/
drwxr-xr-x 2 root root 4096 Sep 17 14:32 logs/
drwxr-xr-x 2 root root 4096 Sep 17 14:17 meta/
root@ubuntu-s-1vcpu-1gb-sgp1-01:~/.small-dragonfly#
```

I am so wired if the document describes wrongly or I perform wrongly.

### Ⅲ. Describe what you expected to happen

The actual behavior should be the same with the document.

### Ⅳ. How to reproduce it (as minimally and precisely as possible)

Just perform every step in that doc.

### Ⅵ. Environment:

- dragonfly version: 0.4.3
- OS (e.g. from /etc/os-release):
- Kernel (e.g. `uname -a`):
- Install tools:
- Others:

**Comment by yeya24** [comment]
Yes, since https://github.com/dragonflyoss/Dragonfly/pull/783, the log file will be rotated regularly. So we cannot find the output in dfclient.log. I think we should add some docs about this behavior. @inoc603

**Comment by fengzixu** [comment]
@yeya24 I think we need to find an approach to show "The file has been downloaded successfully by Dragonfly", not only to describe the behavior of log roration in doc. WDYT?

**Comment by yeya24** [comment]
> @yeya24 I think we need to find an approach to show "The file has been downloaded successfully by Dragonfly", not only to describe the behavior of log roration in doc. WDYT?

I agree with you. Do you have any suggestions? I think a way to do this is via dfdaemon. It can expose some information like api to let uses know if one image is successfully downloaded.  However, this is architecture-related because currently dfdaemon itself is just a proxy.

**Comment by fengzixu** [comment]
I think we need to define and output necessary info about downloading images through Dragonfly.

1. Begin to download
2. Downloading
3. Download successfully or failed

We can output related info in the log. That's a simple way.
But I think if we need to supply an interface or command to check the record of downloading.

For example, we can process `dfclient list records`.

```
{
      "ImageName": "nginx",
      "ImageID": "4sda43rsdfsda",
      "tag": "latest",
      "StartedAt": 2019-09-19,
      "UpdatedAt": 2019-09-20,
}
```

@yeya24 @Starnop @inoc603 WDYT?

**Comment by Starnop** [comment]
also cc/ @lowzj 

@fengzixu Thanks for your good idea.
However, dfdaemon is just a proxy to hijack the http request form container engines such as docker, containerd, etc. So it's hard for dfdaemon  to get the complete info of the image.

**Comment by fengzixu** [comment]
So adding some output to illustrate status of downloading images in logs is an available solution. Is it right? @Starnop

**Comment by yeya24** [comment]
> So adding some output to illustrate status of downloading images in logs is an available solution. Is it right? @Starnop

You mean output this information to the stdout of dfdaemon? I don't think it is readable. When there are multiple dfget requests.

**Comment by Starnop** [comment]
> So adding some output to illustrate status of downloading images in logs is an available solution. Is it right? @Starnop

In fact, there isn't a good idea to solve this problem for me at present.  And I will try to find a  better solution. And please share your solution here if you have any ideas. @fengzixu @yeya24 @inoc603 @lowzj

**Comment by fengzixu** [comment]
> > So adding some output to illustrate status of downloading images in logs is an available solution. Is it right? @Starnop
> 
> You mean output this information to the stdout of dfdaemon? I don't think it is readable. When there are multiple dfget requests.

I'm not sure. I think we need to find a way to illustrate the status of downloading images. For end-users, we can let them check images downloaded by Dragonfly through `docker images`. It means that we need to revise the document of QuickStart. But I don't think that is a good approach. Because of checking image status needs to cross two tools. That is inconvenient.

For developers of Dragonfly, illustrating the status of every downloading image is necessary. So we need to think about it more.

**Comment by yeya24** [comment]
As we have mentioned about combining dfdaemon and dfget to one single binary, I think it is OK to add some apis in dfdaemon later.

**Comment by Starnop** [comment]
@fengzixu  #926 has solved this issue actually. 

> For developers of Dragonfly, illustrating the status of every downloading image is necessary. So we need to think about it more.

I totally agree with you. However, as mentioned above, it's almost impossible to get the image download status for Dragonfly itself. And IMOP, we can make it clear using third tools like `opentracing`.

In addition, dfget will report the metrics to supernode when it success to download a file. So we can get something from supernode:

+ check the log `/home/admin/supernode/logs/dfget.log`
+ check the metrics form supernode API

**Comment by fengzixu** [comment]
@Starnop Just need to revise our doc of QuickStart. :)

**Comment by yeya24** [comment]
> @fengzixu #926 has solved this issue actually.
> 
> > For developers of Dragonfly, illustrating the status of every downloading image is necessary. So we need to think about it more.
> 
> I totally agree with you. However, as mentioned above, it's almost impossible to get the image download status for Dragonfly itself. And IMOP, we can make it clear using third tools like `opentracing`.
> 
> In addition, dfget will report the metrics to supernode when it success to download a file. So we can get something from supernode:
> 
> * check the log `/home/admin/supernode/logs/dfget.log`
> * check the metrics form supernode API

This reminds me about the docs of monitoring dfget. I haven't add it yet. In order to see the logs , supernode must start in debug mode(-D)

**Comment by inoc603** [comment]
@fengzixu will the `analyze` sub command I proposed in #806 meet your needs?

**Comment by inoc603** [comment]
If I remember correctly, dfdaemon only pass layer data to docker daemon when the whole layer is downloaded. When using `docker pull`, progress bar for each layer either stops at zero, or reach 100% in an instant. If we implement some kind of streaming mechanism, it will be more clear that the image is being downloaded.

**Comment by Starnop** [comment]
>  If we implement some kind of streaming mechanism, it will be more clear that the image is being downloaded.

I totally agree with you. In addition, we will not need to  write the file to disk and read it again to  dfdaemon, which will reduce the IO pressure of peer node.
